lib/types.ts

```diff
@@ -85,6 +85,8 @@ export type InsightsEventType = "click" | "conversion" | "view";
 export type InsightsEventConversionSubType = "addToCart" | "purchase";
 
 export type InsightsEventObjectData = {
+  queryID?: string;
+
   price?: number;
```

Comment:
price should also accept a string (and so should discount) in order for callers to be able to provide us with precise decimal numbers, rather than floats.

Same applies to top level `value`.